### ISSUE TITLE
Message Invariant Syntax

### ISSUE THREAD

**Issue by Eric-Edlund** [cubesatlab/merc#11]
The current Merc implementation uses this syntax for message invariant:
```
condition
    :   (WITH INVARIANT RPOINT expression COMMA)* WITH INVARIANT RPOINT expression;
```
which in practice is pretty verbose:
```
message struct <- Name {

} with message_invariant => condition,
  with message_invariant => condition;
```

In the Merc_User_Manual.md I wrote, I described an alternative syntax which I feel is more reasonable:

```
message struct <- Name {

} with condition,  condition;
```

Unless there are reservations about this change, this should be reflected in the Merc implementation. This should be a pretty trivial change, contained in the grammar file and `SymbolTablePopulator`.

**Comment by pchapin** [comment]
The original syntax mimics Ada's aspect syntax, and has the advantage of being more extensible since other characteristics (aspects?) besides "message_invariant" could be added at a later time.

**Comment by pchapin** [comment]
I don't think we had anything in particular in mind. But, for example, imagine a time when Merc can do multiple encodings and not just XDR. I could see something like:

    message struct <- Name {
    }
    with => Encoding("ASN.1"),
            Padding(0),
            AutoReply(True),
            Message_Invariant => some-condition;

As I recall the syntax for the invariant condition was rather simple. It should support conjunctions and disjunctions so that conditions like `(condition1 AND condition2) OR (NOT condition3)` are expressible. I believe this would eliminate the weirdness of using the `message_invariant` multiple times, which I feel like should be an error.

Anyway, the original idea was to borrow the syntax of Ada aspect declarations which are very extensible in ways we can't foresee that we might want later. On the other hand, will we ever actually implement any of these extensions? Probably not.
